lib/src/open_api/spec.dart

```diff
@@ -86,17 +103,63 @@ class OpenApi with _$OpenApi {
   factory OpenApi.fromFile({required String source}) {
     final file = File(source);
     final ext = p.extension(source).toLowerCase();
-    dynamic raw;
-    if (ext.contains('json')) {
-      raw = json.decode((file.readAsStringSync()));
-    } else if (ext.contains('yaml')) {
-      raw = yaml.loadYaml((file.readAsStringSync()));
-      raw = json.decode(json.encode(raw));
-    } else {
+
+    final format = OpenApiFormat.fromExtention(ext);
+    if (format == null) {
       throw Exception('Unsupported file type: $ext');
     }
 
-    return OpenApi.fromJson(Map<String, dynamic>.from(raw));
+    return OpenApi.fromString(
+      source: file.readAsStringSync(),
+      format: format,
+    );
+  }
+
+  // ------------------------------------------
+  // FACTORY: OpenApi.fromString
+  // ------------------------------------------
+
+  /// Create an [OpenApi] object from a JSON/YAML string
+  /// The format will be inferred from the file extension
+  /// When the format is not provided, we will try both JSON and YAML
+  factory OpenApi.fromString(
+      {required String source, required OpenApiFormat? format}) {
```

Comment:
Please add a trailing comma after `format` so that this formats properly, e.g:

```dart
  factory OpenApi.fromString({
    required String source,
    required OpenApiFormat? format,
  }) {
```